src/charm.py

```diff
@@ -8,18 +8,136 @@
 
 import logging
 
-from ops.charm import CharmBase
+from charms.observability_libs.v0.kubernetes_service_patch import KubernetesServicePatch
+from jinja2 import Template
+from ops.charm import CharmBase, PebbleReadyEvent
 from ops.main import main
+from ops.model import ActiveStatus, BlockedStatus, MaintenanceStatus, ModelError, WaitingStatus
+from ops.pebble import ChangeError, Layer
 
 logger = logging.getLogger(__name__)
 
+OATHKEEPER_API_PORT = 4456
+
 
 class OathkeeperCharm(CharmBase):
     """Charmed Ory Oathkeeper."""
 
     def __init__(self, *args):
         super().__init__(*args)
 
+        self._container_name = "oathkeeper"
+        self._container = self.unit.get_container(self._container_name)
+        self._oathkeeper_config_path = "/etc/config/oathkeeper.yaml"
+        self._oathkeeper_access_rules_path = "/etc/config/oathkeeper/access-rules.yaml"
+        self._name = self.model.app.name
+
+        self._kratos_login_url = (
+            f"http://kratos.{self.model.name}.svc.cluster.local:4433/self-service/login/browser"
+        )
+        self._kratos_session_url = (
+            f"http://kratos.{self.model.name}.svc.cluster.local:4433/sessions/whoami"
+        )
+
+        self.service_patcher = KubernetesServicePatch(
+            self, [("oathkeeper-api", OATHKEEPER_API_PORT)]
+        )
+
+        self.framework.observe(self.on.oathkeeper_pebble_ready, self._on_oathkeeper_pebble_ready)
+
+    @property
+    def _oathkeeper_layer(self) -> Layer:
+        """Returns a pre-configured Pebble layer."""
+        layer_config = {
+            "summary": "oathkeeper-operator layer",
+            "description": "pebble config layer for oathkeeper-operator",
+            "services": {
+                self._container_name: {
```

Comment:
Let's not use the `_container_name` for the service name (even if it has the same value let's use a different variable to avoid confusion)